src/user.py

```diff
@@ -0,0 +1,173 @@
+import re
+import uuid
+from datetime import datetime
+
+import validators
+from flasgger import swag_from
+from flask import Blueprint, jsonify, request
+from flask_jwt_extended.utils import get_jwt_identity
+from werkzeug.security import check_password_hash, generate_password_hash
+
+from src.google import get_user_info
+from src.models import UserLogin, UserProfile, db
+from src.constants.http_status_codes import HTTP_200_OK, HTTP_201_CREATED, HTTP_400_BAD_REQUEST, HTTP_401_UNAUTHORIZED, HTTP_409_CONFLIT
+
+
+
+user_bp = Blueprint('user', __name__, url_prefix='/api')
+
+
+@user_bp.post('/login/user/signup')
+@swag_from('./docs/signup/user.yml')
+def signup():
+    """Endpoint for user signup
+
+    Returns:
+        json: user's information
+    """
+    
+    if 'first_name' not in request.json:
+        return jsonify({
+            'error': "first_name is missing from request"
+        }), HTTP_400_BAD_REQUEST
+        
+    if 'last_name' not in request.json:
+        return jsonify({
+            'error': "last_name is missing from request"
+        }), HTTP_400_BAD_REQUEST
+    
+    if 'email' not in request.json:
+        return jsonify({
+            'error': "email is missing from request"
+        }), HTTP_400_BAD_REQUEST
+    
+    if 'password' not in request.json:
+        return jsonify({
+            'error': "password is missing from request"
+        }), HTTP_400_BAD_REQUEST
+        
+    first_name = request.json['first_name']
+    last_name = request.json['last_name']
+    email = request.json['email']
+    password = request.json['password']
+    
+    if not validators.email(email):
+        return jsonify({
+            'error': "email is invalid"
+        }), HTTP_400_BAD_REQUEST
+        
+    if not validate_password_complexity(password):
+        return jsonify({
+            'error': "password should be at least 8 characters long and it contain minimum one lower case, one uppercase and one special character"
```

Comment:
`password should be at least 8 characters long and must contain minimum one lower case, one uppercase and one special character`